### DISCUSSION TITLE
Using git-lfs as a library

### DISCUSSION THREAD

**Discussion by zendril** [git-lfs/git-lfs#5646] [Q&A]
It appears that this codebase is mainly structured to be used as cli/executable.

Can it be used as a client library inside my go app? 
I'm effectively trying to do a `git lfs fetch -all`
I'm assuming this can be done, but don't want to go down a rabbit hole attempting it if there is something that would prevent that.
I also have had very little experience with git lfs in general, so maybe I'm overthinking it and there is a simple way to achieve what I'm asking w/o even pulling in a library.

Has anyone done this or have some examples?

Thanks!

**Comment by bk2204** [comment]
Hey,

We don't provide a stable non-CLI API, so while you are free under the license to use it as a library, it may break in incompatible ways between non-major versions.  (In fact, that is nearly guaranteed to happen between minor versions.)  You may wish to simply invoke the binary using `os/exec` in the standard library if that meets your needs.

We've discussed providing a C-compatible library (written in Rust) in the future, but we don't have any concrete plans to build that at the moment.

**Comment by zendril** [comment]
Pretty much what I expected to be the case, and in my scenario I'm trying to avoid having to have the other executable on the filesystem and exec to it. Appreciate the quick response and thinking around future plans.

cheers
